### PR TITLE
Main update to 0.1.2

### PR DESCRIPTION
**Pull request by JakePost**
Changelog based on PRs:

- Fixed bug in eccentricity functions. More catches for orb_a > disk_radius_outer. [#256](https://github.com/McFACTS/McFACTS/pull/256)
- Add unit test and pytest workflow [#255](https://github.com/McFACTS/McFACTS/pull/255) [#254](https://github.com/McFACTS/McFACTS/pull/254)
- Add terminal text after setup command, fix python version requirements [#249](https://github.com/McFACTS/McFACTS/pull/249)
- Fixed bug in excluded_angles in bin_spheroid_encounter [#251](https://github.com/McFACTS/McFACTS/pull/251)
- Fixing bugs in type1_migration and retro_bh_orb_disk_evolve [#250](https://github.com/McFACTS/McFACTS/pull/250)
- All functions converted to object-orientated, bug fixes for migration [#247](https://github.com/McFACTS/McFACTS/pull/247)
- Updated IOdocumentation and added a flag variable to enable galaxy runs directory generation [#243](https://github.com/McFACTS/McFACTS/pull/243)
- evolve.bin_reality_check now checks more things (such as ionization due to eccentricity > 1), removing the need for a separate reality check for this in mcfacts_sim.py
- Structural/physical edits in dynamics (circular_binaries_encounters_ecc_prograde, circular_binaries_encounters_circ_prograde)
- New eccentricity for each component of an ionized binary estimated in new way

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -53,9 +53,15 @@ setup: clean version
 	source ~/.bash_profile && \
 	conda activate base && \
 	conda remove -n mcfacts-dev --all -y && \
-	conda create --name mcfacts-dev "python>=3.10.4<=3.13" pip "pytest" -c conda-forge -c defaults -y && \
+	conda create --name mcfacts-dev "python>=3.10.4,<=3.13" pip "pytest" -c conda-forge -c defaults -y && \
```
Comment (xevra) [inline Makefile]: Should be <3.13

3.13 does not work right now with the default conda installation.

File: README.md [hunk 2]
```diff
@@ -130,6 +132,11 @@ McFACTS will also generate the following plots:
 
 Paper 1: (https://arxiv.org/abs/2410.16515)
 
-McFACTS is developed by Barry McKernan, K. E. Saavik Ford, Harrison E. Cook, Vera Delfavero, Kaila Nathaniel, Jake Postiglione, Shawn Ray, and Richard O'Shaughnessy. If you use McFACTS in your work, please cite the following:
+Paper 2: *In Preparation*
```
Comment (xevra) [inline README.md]: https://arxiv.org/abs/2411.10590

File: README.md [hunk 3]
```diff
@@ -130,6 +132,11 @@ McFACTS will also generate the following plots:
...
+
+Paper 3: (https://arxiv.org/abs/2410.18815)
+
+McFACTS is developed by Barry McKernan, K. E. Saavik Ford, Harrison E. Cook, Vera Delfavero, Emily McPike, Kaila Nathaniel, Jake Postiglione, Shawn Ray, and Richard O'Shaughnessy. If you use McFACTS in your work, please cite the following:
 
-* McKernan, B., Ford, K.E.S., Cook, H.E., et al., *McFACTS I: Testing the LVK AGN channel with Monte Carlo for AGN Channel Testing and Simulation (McFACTS)*. 2024. ([arXiv](https://arxiv.org/abs/2410.16515))
\ No newline at end of file
+* McKernan, B., Ford, K.E.S., Cook, H.E., et al., *McFACTS I: Testing the LVK AGN channel with Monte Carlo for AGN Channel Testing and Simulation (McFACTS)*. 2024. ([arXiv](https://arxiv.org/abs/2410.16515))
```
Comment (xevra) [inline README.md:141]: Cook, H.  E., et al., *McFACTS II: Mass Ratio--Effective Spin Relationship of Black Hole Mergers in the AGN Channel*. 2024. ([arXiv](https://arxiv.org/abs/2411.10590))

File: src/mcfacts/external/DiskModelsPAGN.py [hunk 4]
```diff
@@ -223,11 +223,11 @@ def dSigmadR(obj):
...
     """
-    Sigma = 2*obj.rho*obj.h # descrete
-    rlog10 = np.log10(obj.R)  # descrete
-    Sigmalog10 = np.log10(Sigma)  # descrete
-    Sigmalog10_spline = UnivariateSpline(rlog10, Sigmalog10, k=3, s=0.005, ext=0)  # need scipy ver 1.10.0
-    dSigmadR_spline =  Sigmalog10_spline.derivative()
+    Sigma = 2*obj.rho*obj.h  # discrete
+    rlog10 = np.log10(obj.R)  # discrete
+    Sigmalog10 = np.log10(Sigma)  # discrete
+    Sigmalog10_spline = scipy.interpolate.UnivariateSpline(rlog10, Sigmalog10, k=3, s=0.005, ext=0)  # need scipy ver 1.10.0
```
Comment (xevra) [inline src/mcfacts/external/DiskModelsPAGN.py:229]: Is there a reason we are not using Cubic Splines?

**Comment by xevra** [comment]
Items to add to changelog:
- evolve.bin_reality_check now checks more things (such as ionization due to eccentricity > 1), removing the need for a separate reality check for this in mcfacts_sim.py
- structural/physical edits in dynamics (circular_binaries_encounters_ecc_prograde, circular_binaries_encounters_circ_prograde)
- New eccentricity for each component of an ionized binary estimated in new way

**Comment by xevra** [comment]
I'm not really sure that a *granular changelog* is possible right now, considering thousands of lines were changed in this version, so I think adding these points will be sufficient.

**Review by xevra — APPROVED** [review]
I did make some comments, for things that I think could be changed in the setup, changelog, and documentation. However, these are minor, I don't think these things should hold back the merge.

**Comment by JakePost** [comment]
Added a changelog file, and updated the README and Makefile with the changes you pointed out @xevra  😃 

I think the Cubic vs Univariant spline might be something to tackle in a future PR / Issue?

**Review by bmckernan — APPROVED** [review]
Agreed!

**Pull request merged**